lib/moonshot/build_mechanism/script.rb

```diff
@@ -41,7 +41,7 @@ def build_hook(version)
 
   def post_build_hook(_version)
     unless File.exist?(@output_file) # rubocop:disable GuardClause
-      raise Thor::Error, 'Build command did not produce output file!'
+      step.failure 'Build command did not produce output file!'
```

Comment:
This is how @jamesiarmes handled that.

https://github.com/acquia/moonshot/blob/master/lib/moonshot/controller.rb#L67-L78

Perhaps we want to consider how we can do that more generally with the hooks.